### PR TITLE
change torch.cuda.amp.GradScaler to torch.GradScaler("cuda")

### PR DESCRIPTION
**Pull request by 0ssamaak0**
## Description
using `torch.cuda.amp.GradScaler` will result in a warning
```
FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
```
So it should be changed according to [Automatic Mixed Precision package Documentation](https://pytorch.org/docs/stable/amp.html#gradient-scaling) `torch.cuda.amp.GradScaler` to `torch.GradScaler("cuda")` to match

## Checklist

- [ ] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

### CODE REVIEW

File: recipes_source/recipes/amp_recipe.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def make_model(in_size, out_size, num_layers):
 # The same ``GradScaler`` instance should be used for the entire convergence run.
 # If you perform multiple convergence runs in the same script, each run should use
 # a dedicated fresh ``GradScaler`` instance. ``GradScaler`` instances are lightweight.
-scaler = torch.cuda.amp.GradScaler()
+scaler = torch.GradScaler("cuda")
```
Comment (spzala) [inline recipes_source/recipes/amp_recipe.py:153]: @0ssamaak0 thanks for the PR. The deprecated warning says to use `torch.amp.GradScaler('cuda', args...)` so why are you changing to `torch.GradScaler("cuda")` ?  i.e. why did you remove `amp`

File: recipes_source/recipes/amp_recipe.py [hunk 2]
```diff
@@ -182,7 +182,7 @@ def make_model(in_size, out_size, num_layers):
 
 net = make_model(in_size, out_size, num_layers)
 opt = torch.optim.SGD(net.parameters(), lr=0.001)
-scaler = torch.cuda.amp.GradScaler(enabled=use_amp)
+scaler = torch.GradScaler("cuda" ,enabled=use_amp)
```
Comment (spzala) [inline recipes_source/recipes/amp_recipe.py:185]: Same as above.

**Review by spzala — COMMENTED** [review]
Please see comments inline. Thanks!

File: recipes_source/recipes/amp_recipe.py (hunk 1 above)
Comment (0ssamaak0) [reply recipes_source/recipes/amp_recipe.py:153]: I followed this
image

[Source](https://pytorch.org/docs/stable/amp.html#gradient-scaling)

File: recipes_source/recipes/amp_recipe.py (hunk 2 above)
Comment (0ssamaak0) [reply recipes_source/recipes/amp_recipe.py:185]: I followed this
image

[Source](https://pytorch.org/docs/stable/amp.html#gradient-scaling)

File: recipes_source/recipes/amp_recipe.py (hunk 1 above)
Comment (spzala) [reply recipes_source/recipes/amp_recipe.py:153]: Thanks for the quick reply. `GradScaler` can be used on its own but because you are modifying `amp` recipe, I would suggest to use `torch.amp.GradScaler` as it's also mentioned in the code - https://github.com/pytorch/pytorch/blob/main/torch/cuda/amp/grad_scaler.py#L12 
Also, if you run the recipe, you will notice the warning message suggesting to use `torch.amp.GradScaler`

File: recipes_source/recipes/amp_recipe.py (hunk 2 above)
Comment (spzala) [reply recipes_source/recipes/amp_recipe.py:185]: Thanks for the quick reply. `GradScaler` can be used on its own but because you are modifying `amp` recipe, I would suggest to use `torch.amp.GradScaler` as it's also mentioned in the code - https://github.com/pytorch/pytorch/blob/main/torch/cuda/amp/grad_scaler.py#L12 
Also, if you run the recipe, you will notice the warning message suggesting to use `torch.amp.GradScaler`

Comment (0ssamaak0) [reply recipes_source/recipes/amp_recipe.py]: I have changed the code accordingly.

Thank you

Comment (spzala) [reply recipes_source/recipes/amp_recipe.py]: Thanks!

**Review by malfet — APPROVED** [review]
Looks good for me, thank you for the update
